pkg/api/auth_middleware.go

```diff
@@ -148,11 +160,107 @@ func enhanceWithFriendlyName(user *model.User, friendlyName string) *model.User
 	return user
 }
 
+// userFromSAML returns a user from an existing SAML session.
+// If the user doesn't exist on the lakeFS side, it is created.
+// This function does not make any calls to an external provider.
+func userFromSAML(ctx context.Context, logger logging.Logger, authService auth.Service, authSession *sessions.Session, cookieAuthConfig *config.CookieAuthVerification) (*model.User, error) {
+	idTokenClaims, ok := authSession.Values[SAMLTokenClaimsSessionKey].(oidc_encoding.Claims)
+
+	if idTokenClaims == nil {
+		return nil, nil
+	}
+
+	if !ok {
+		logger.WithField("claims", authSession.Values[SAMLTokenClaimsSessionKey]).Error("failed decoding tokens")
+		return nil, ErrAuthenticatingRequest
+	}
+
+	logger.WithField("claims", idTokenClaims).Debug("success decoding token claims")
+
+	idKey := cookieAuthConfig.ExternalUserIDClaimName
+	externalID, ok := idTokenClaims[idKey].(string)
+	if !ok {
+		logger.WithField(idKey, idTokenClaims[idKey]).Error("failed type assertion for sub claim")
+		return nil, ErrAuthenticatingRequest
+	}
+
+	log := logger.WithField("external_id", externalID)
+
+	for claimName, expectedValue := range cookieAuthConfig.ValidateIDTokenClaims {
```

Comment:
Who defines what's expected?

Reply:
The user in the configuration, same as OIDC where we can additional validate claims like `department` , `organization` etc. 

A User might configure the external system to send more claims.